### PR TITLE
NEWS and README updates

### PR DESCRIPTION
**Pull request by hppritcha**
JMS note some questions (grep for JMS).

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

[skip ci]
bot:notest

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -620,13 +620,13 @@ Network Support
...
-    - Mellanox MXM
     - Portals4
     - OpenFabrics Interfaces ("libfabric" tag matching)
 
     Open MPI will, by default, choose to use "cm" when one of the
-    above transports can be used.  Otherwise, "ob1" will be used and
-    the corresponding BTLs will be selected. Users can force the use
+    above transports can be used, unless OpenUCX or MXM support is detected,
+    in which case the "ucx" or "yalla" the PML will be used by default.  Otherwise,
+    "ob1" will be used and the corresponding BTLs will be selected. Users can force the use
```
Comment (jsquyres) [inline README]: Need to re-flow this paragraph.

File: NEWS [hunk 2]
```diff
@@ -98,8 +101,9 @@ Bug fixes/minor improvements:
     "file:filename").
   - orte_timeout_for_stack_trace: number of seconds to wait for stack
     traces to be reported (or <=0 to wait forever).
-- Various improvements to the Portals 4 MTL, to include adding support
-  for non-contiguous datatypes.
+  - mtl_ofi_control_prog_type/mtl_ofi_data_prog_type: specify libfabric
+    progress model to be used for control and data
+- Add support for non-contiguous datatypes to the Portals 4 OSC.
```
Comment (jsquyres) [inline NEWS]: Should probably say "one-sided component" here (I don't think "OSC" is a well-known acronym outside of the developer community).

File: NEWS [hunk 3]
```diff
@@ -98,8 +101,9 @@ Bug fixes/minor improvements:
     "file:filename").
   - orte_timeout_for_stack_trace: number of seconds to wait for stack
     traces to be reported (or <=0 to wait forever).
-- Various improvements to the Portals 4 MTL, to include adding support
-  for non-contiguous datatypes.
+  - mtl_ofi_control_prog_type/mtl_ofi_data_prog_type: specify libfabric
+    progress model to be used for control and data
```
Comment (jsquyres) [inline NEWS]: Missing period at end of sentence.

File: NEWS [hunk 4]
```diff
@@ -118,13 +122,16 @@ Bug fixes/minor improvements:
   orphan children nodes in the launch tree.
 - Fix the help message when showing deprecated MCA param names to show
   the correct (i.e., deprecated) name.
+- Enable support for OpenIB BTL use across different IB subnets.
```
Comment (jsquyres) [inline NEWS]: I think we more-or-less uniformly refer to it as the "openib" BTL (lower case).

File: NEWS [hunk 5]
```diff
@@ -75,6 +75,9 @@ JMS Any other major new features to list?
 
 Changes in behavior compared to prior versions:
 
+- Added a timeout option to mpirun command line: number of
+  seconds for mpirun to wait for a job to complete
```
Comment (jsquyres) [inline NEWS]: I think that this is a minor new feature, not a new behavior.

File: NEWS [hunk 6]
```diff
@@ -118,13 +122,16 @@ Bug fixes/minor improvements:
...
+- Enable support for OpenIB BTL use across different IB subnets.
 - Fix a minor error in MPI_AINT_DIFF.
 - Fix bugs with MPI_IN_PLACE handling in:
   - MPI_ALLGATHER[V]
   - MPI_IALLTOALL*
   - MPI_[I][GATHER|SCATTER][V]
   - MPI_IREDUCE[_SCATTER]
   - Thanks to all the users who helped diagnose these issues.
+  JMS - WERE ANY OF THESE NOT FIXED in 2.0.X? See below for numerous
+  citations.
```
Comment (jsquyres) [inline NEWS]: Good catch; I missed the de-duplication.  I think ALLGATHER[V] and IALLTOALL* were cited below, but I didn't see [I][GATHER|SCATTER][V] or IREDUCE[_SCATTER].

**Comment by jsquyres** [comment]
See https://github.com/hppritcha/ompi/pull/1.

**Comment by jsquyres** [comment]
@hppritcha I assume you are going to squash all 3 of those commits into 1.

**Comment by jsquyres** [comment]
Howard: I merged a minor NEWS update via https://github.com/open-mpi/ompi/pull/3068 which may cause conflicts with this PR (i.e., you may need to rebase on top of #3068 first).  Sorry!  ☹️

**Comment by hppritcha** [comment]
@jsquyres want to take a final look (still need to update the date in NEWS)

**Pull request merged**